aikau/src/main/resources/alfresco/services/SiteService.js

```diff
@@ -313,11 +313,52 @@ define(["dojo/_base/declare",
       },
 
       /**
+       * This returns all sites that the user has the permission to see. Eg. all public sites.
        *
        * @instance
        * @param {object} payload The details of the request
        */
       getSites: function alfresco_services_SiteService__getSites(payload) {
+        var url = AlfConstants.PROXY_URI + "api/sites";
+        var hasArguments = false;
+        //Add name filter if supplied
+        if (payload.nameFilter) {
+          if (hasArguments) {
```

Comment:
I believe the code could be optimised before being merged by using the ```alfresco/util/urlUtils``` module to handle the URL construction. Specifically the ```addFilterQueryParameters``` operation can already deal with an arbitrary number of ```payload.dataFilter``` entries to be added to the URL.